### PR TITLE
Delegate parsing of targetFramework to dotnet CLI

### LINKED ISSUE #3565: `NullReferenceException` when `TargetFramework` is only specified in `Directory.Build.props`
Exception when running `func azure functionapp publish`:

```
   at Azure.Functions.Cli.Actions.AzureActions.PublishFunctionAppAction.<RunAsync>d__66.MoveNext() in Azure.Functions.Cli.Actions.AzureActions\PublishFunctionAppAction.cs:line 179
```

### PR DESCRIPTION
**Pull request by gukoff**
### Issue describing the changes in this PR

Instead of parsing the .csproj file, determine the target framework by calling `dotnet build -getproperty:TargetFramework`.

It "works on my machine" (OSX, dotnet 8.0.201). I'm not aware if this solution has portability issues.

resolves #3565
 
#### Alternatives considered

I tried using `new Project(path)` instead of `ProjectRootElement.Open(path)` in hopes that `new Project` would find and import any external properties.

It didn't run. My .csproj file starts with `<Project Sdk="Microsoft.NET.Sdk">`, and calling `new Project(csproj_content)` throws exception `The SDK 'Microsoft.NET.Sdk' specified could not be found`.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [?] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by jviau** [comment]
@gukoff I believe one error you will encounter is welcome message for new dotnet SDK installs. You will need to add some environment variables before executing the dotnet command:

```
DOTNET_NOLOGO = 1
DOTNET_SKIP_FIRST_TIME_EXPERIENCE = 1
DOTNET_CLI_TELEMETRY_OPTOUT = 1
```

File: src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs [hunk 1]
```diff
@@ -25,6 +27,28 @@ public static void EnsureDotnet()
...
+        /// Function that determines TargetFramework of a project even when it's defined outside of the .csproj file,
+        /// e.g. in Directory.Build.props
+        /// </summary>
+        /// <param name="projectDirectory">Directory containing the .csproj file</param>
+        /// <returns>Target framework, e.g. net8.0</returns>
+        /// <exception cref="CliException"></exception>
+        public static async Task<string> DetermineTargetFramework(string projectDirectory)
+        {
+            EnsureDotnet();
+            var exe = new Executable("dotnet", "build -getproperty:TargetFramework", workingDirectory: projectDirectory);
```
Comment (kshyju) [inline src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs:40]: Thanks @gukoff for the contribution.

The change looks good. Would you mind addressing  [this comment](https://github.com/Azure/azure-functions-core-tools/pull/3604#issuecomment-2221428606) by @jviau?  I think no-logo can be passed in the command itself `dotnet build -getproperty:TargetFramework --nologo` . The other 2,we may need to set via env variables for this command (and may need to reset to previous state after this command is executed)

Comment (gukoff) [reply src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs]: Good idea @jviau, I added the env vars.

One caveat: I haven't worked on this project for a while, and I don't remember how to test these changes :(

**Review by kshyju — DISMISSED** [review]
Thanks @gukoff 
Looks good. Some minor comments.

File: src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs [hunk 2]
```diff
@@ -25,7 +27,39 @@ public static void EnsureDotnet()
...
+        {
+            EnsureDotnet();
+            var exe = new Executable(
+                "dotnet",
+                "build -getproperty:TargetFramework",
+                workingDirectory: projectDirectory,
+                environmentVariables: new Dictionary<string, string>
+                {
+                    // https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables
+                    ["DOTNET_SKIP_FIRST_TIME_EXPERIENCE"] = "1",  // do not write disclaimer to stdout
```
[the remark this answers is not in the document]
Comment (gukoff) [reply src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs:47]: Makes sense. As far as I understand, even 3.0 [is not a supported runtime](https://learn.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=isolated-process%2Cv4&pivots=programming-language-csharp#languages) anymore.

**Pull request merged**